Doc 12022 spm instructions fix

ElliotFrancisHunter opened the pull request:
Fixed SPM instructions referencing incorrect version (3.1.6 instead of 3.2.0-beta.1)

Caused by incorrect attribute values used.

Expanded scope on the logic that now the attribute was fixed it should be applied to the pages and modules that required it.
Also did some fly-bys

-  Updated minimum API version on android to 23
-  Changed 'Beta 1' text to just 'Beta' as decision was made to change it on docs site rather than beta 1,2,3 as there is no need to have separate pages for the versions.
-  Added attributes for vector search for all sdks to remove hardcoded version for that too as this will be useful for subsequent vector search releases and was needed anyway.
- Changed hardcoded tag attribute in swift, android and objc navs to an include of the relevant module component 
-  Added a previous-version attribute to antora.yml so we don't get a 403 on api reference docs for mobile sdks yet to have a 3.2.0 beta release

ElliotFrancisHunter committed:
DOC-12022: Fix spm referencing incorrect package version

Changes:
. Add releasetag definition to definecomponentattributes root page
. Add attributes for vector search versions in antora.yml to track different versions across mobile sdks
. Add vector search specific attributes to definemoduleattributes pages to remove hardcoded versions
. Removed hardcoded '-beta.1'

Issues:
. API reference link on swift nav is currently returning 403

ElliotFrancisHunter committed:
DOC-12022: Apply attribute changes to all vector search supported sdks

Also removed hardcode '-beta.1's.
Android and objc

ElliotFrancisHunter committed:
DOC-12022

Added missing minimum android API update.

osfameron commented on modules/ROOT/pages/_partials/_define_component_attributes.adoc:
```diff
@@ -8,7 +8,11 @@ ifdef::is_diag[_define_component_attributes.adoc]
+// Basic version for Vector Search
+:vs-version: {vs-major}.{vs-minor}
 // Base three digit version including optional tag
+:releasetag: beta.1
```
I'd move this declaration to the antora.yml

e.g. The antora.yml is for things that we *expect to change* at a release, like configuration.

The partial is more like plumbing, to make sure that the things declared in the configuration get formatted in the right way.

osfameron replied:
not sure that will magically fix it... but it's right to move it to antora.yml, so try that and see?
If not, I'll spend some more time on this tomorrow! Not sure what the problem is from a quick scan.

osfameron commented on modules/ROOT/pages/_partials/_define_component_attributes.adoc:
```diff
@@ -18,11 +22,20 @@ ifdef::releasetag[:tag: -{releasetag}]
-:version-maintenance-net: {major}.{minor}.{maintenance-net}
-:version-maintenance-java: {major}.{minor}.{maintenance-java}
-:version-maintenance-ios: {major}.{minor}.{maintenance-ios}
+:version-maintenance-android: {major}.{minor}.{maintenance-android}{tag}
```
So, I don't know why these didn't *already* have {tag}...
but as this is a new addition, we'll need to test that this hasn't broken some behavious elsewhere.

ElliotFrancisHunter replied:
It's only used in install, the definition itself and quickstart. Have tested and verified output is correct in all locations.

ElliotFrancisHunter committed:
DOC-12022: Apply feedback and fixes

Added:
. Moved releasetag to antora.yml so we don't need instances of it in every module
. Added a previous-release attribute to antora.yml as a quick fix to ensure API reference docs don't return a 403 for Java, C and .Net.

Comments:
. Could probably add previous-version to Java, C and .Net docs where appropriate but it seems like a massive effort for something that will resolve itself in a couple of months and could risk more bugs due to human error.

ElliotFrancisHunter committed:
DOC-12022: Remove local attributes

Use version-maintenance-[platform] derived from definecomponentattribute file instead of local varaibles

osfameron commented on modules/android/nav-android.adoc:
```diff
@@ -1,5 +1,4 @@
-:tag: -beta.1
-ifdef::releasetag[:tag: -{releasetag}]
+include::partial$_set_page_context_for_android.adoc[]
```
You know, I wonder if this line is required?
Given that on the older version of line 48 below, {major} and {minor} etc. *were resolved*...
hmm, I guess those came from the Antora.yml, rather than the page context...

OK, never mind (but I'll try to remember to play with it at some point...)

osfameron commented on modules/android/pages/_partials/_define_module_attributes.adoc:
```diff
@@ -6,6 +6,12 @@
 :version-maintenance-hyphenated: {major}-{minor}-{maintenance-android}{tag}
+
+// VECTOR SEARCH attributes
+:vs-version-maintenance: {vs-version}.{vs-maintenance-android}{tag}
```
It's a bit odd that you use {vs-version} here, but {vs-major}-{vs-minor} directly below.

ElliotFrancisHunter replied:
It's the hyphens and to mirror the hyphenated version above. I'm not too sure *when* exactly this is used as it's probably an edge case but added it in the off chance I'll need it at some point.

osfameron reviewed (approved):
Looks good!
